libs/ojsonnet/Core_jsonnet.ml

```diff
@@ -161,11 +161,65 @@ and for_comp = tok (* 'for' *) * ident * tok (* 'in' *) * expr
 [@@deriving show { with_path = false }]
 
 (*****************************************************************************)
-(* Program *)
+(* Env *)
 (*****************************************************************************)
+type env = {
```

Comment:
Why move those types to Core_jsonnet.ml? We could keep Value_jsonnet.ml and move the environment that was in Eval_jsonnet.ml in Value_jsonnet.ml.

I try hard to "separate concerns" and so define data-structures in different files if they are quite independent.
For instance we have some phases like Desugar_jsonnet.ml that can produce Core_jsonnet.program and this phase does not have to know about environment and values.

Reply:
I think the reason I moved these originally, is I tried to keep environment in a separate file, which then lead to some mutual recursion issues, so I ended up moving everything into 1. But, yes! Moving env into Value_jsonnet works, will fix  that.